table/tables/tables.go

```diff
@@ -1091,8 +1091,9 @@ func (t *TableCommon) Allocators(ctx sessionctx.Context) autoid.Allocators {
 }
 
 // RebaseAutoID implements table.Table RebaseAutoID interface.
-func (t *TableCommon) RebaseAutoID(ctx sessionctx.Context, newBase int64, isSetStep bool) error {
-	return t.Allocators(ctx).Get(autoid.RowIDAllocType).Rebase(t.tableID, newBase, isSetStep)
+// Both auto-increment and auto-random can use this function to do rebase on explicit newBase value (without shadow bits).
+func (t *TableCommon) RebaseAutoID(ctx sessionctx.Context, newBase int64, isSetStep bool, tp autoid.AllocatorType) error {
```

Comment:
Please add UT for the type of `autoid.AutoRandomType`.

Reply:
yes, we have already had.
`alter table auto_random_rebase` will trigger it -> `onRebaseAutoRandomType` -> `onRebaseAutoID`

Reply:
I mean UT in tables_test.go like https://github.com/pingcap/tidb/pull/16750/files#diff-b8866f79039da1517d6fe97d7c2d7d4cR164.